betterconf/caster.py

```diff
@@ -46,12 +48,12 @@ class BoolCaster(ConstantCaster):
 
 
 class IntCaster(AbstractCaster):
-    def cast(self, val: str) -> typing.Union[str, int]:
+    def cast(self, val: str) -> typing.Union[int, ImpossibleToCastError]:
         try:
             as_int = int(val)
             return as_int
         except ValueError:
-            return val
+            return ImpossibleToCastError(val, self)
```

Comment:
The same thing: I feel unnatural to see that code can `return` a error.
I think it's a good practice in languages like Go, but Python has robust exception handling and try-except, why not use that?

Reply:
To clarify: if you aren't familiar with Go (neither am I), it has lots of things like this:
```go
if err != nil {
    // do something
}
```